lens-cube/src/main/java/org/apache/lens/cube/parse/JoinCandidate.java

```diff
@@ -0,0 +1,117 @@
+package org.apache.lens.cube.parse;
+
+import java.util.*;
+
+import org.apache.lens.cube.metadata.FactPartition;
+import org.apache.lens.cube.metadata.TimeRange;
+import org.apache.lens.server.api.error.LensException;
+
+/**
+ * Represents a join of two candidates
+ */
+public class JoinCandidate implements Candidate {
+
+  /**
+   * Child candidates that will participate in the join
+   */
+  private Candidate childCandidate1;
+  private Candidate childCandidate2;
+  private String toStr;
+  private QueryAST queryAST;
+  private CubeQueryContext cubeql;
+
+  public JoinCandidate(Candidate childCandidate1, Candidate childCandidate2, CubeQueryContext cubeql) {
```

Comment:
Is there any particular reason why we are limiting joins to 2 candidates and not allowing an arbitrary number of candidates to be joined.